### ISSUE TITLE
Virtual scrolling doesn't work if Handsontable's container is a virtual element.

### ISSUE THREAD

**Issue by mrpiotr-dev** [handsontable/handsontable#5633]
### Description
Full issue's background was described in handsontable/angular-handsontable#43.

To make a long story short, if you try to initialize Handsontable in a virtual created element, then virtual scrolling doesn't work.

### Steps to reproduce
1. Open demo.
2. Scroll vertically/horizontally.

* There is a workaround - you can use `this.updateSettings({})` instead of `this.render()`.

### Demo
https://jsfiddle.net/mcvqyrd1/

### Your environment
* Handsontable version: 1.14.0 (0.34.0)+
* Browser Name and version: Chrome, Safari, Edge

**Comment by jeremy-smith-maco** [comment]
If you use `hot.view.wt.wtOverlays.updateMainScrollableElements()` after initializing the Handsontable that also works (not inside the `afterInit` callback). Not sure if that information helps or not.

**Comment by naveenjaiswal** [comment]
ETA for resolution of this issue ?

**Comment by AMBudnik** [comment]
Hey @naveenjaiswal 

the fix has been postponed. We need some time to figure out the exact approach. I will send you a notification once we get this done.

**Comment by AMBudnik** [comment]
Handsontable v12.1.2 works the same https://jsfiddle.net/jrq0aphc/

![Screenshot 2022-08-29 at 11 58 12](https://user-images.githubusercontent.com/10757813/187175612-992ed69f-ccd8-44a0-a949-b286f249eb3a.png)

![Screenshot 2022-08-29 at 11 58 48](https://user-images.githubusercontent.com/10757813/187175722-4ac1daa2-c53f-4c85-8468-596928b2bf21.png)

**Comment by budnix** [comment]
Hi @naveenjaiswal, @jeremy-smith-maco,

This is now fixed. The fix is in **Handsontable 18.1.0**, released on September 1, 2026.

### What was wrong

Handsontable decided which element scrolls the grid only once, while the grid was being created. An element that is not in the page yet has no layout, so the browser reports no styles for it or for any of its parents. Reading those styles found nothing that scrolls, so Handsontable fell back to the window. Once the element was added to the page, that decision was never made again — it stayed wrong for as long as the instance lived.

That one wrong decision caused three separate symptoms:

- Scrolling the grid rendered no new rows or columns.
- Column headers and frozen columns stopped moving with the body.
- Virtualization was effectively off, so the whole data set rendered at once.

### What was changed

Handsontable now marks that decision as temporary when the grid is built without a layout, and makes it again after the first draw that finds the grid in the page. The fix landed in #13208.

### What I tested

I ran the demo from this issue on three versions, and I also tested two other ways of adding the container to the page, because frameworks do it in different ways:

| Version | Demo from this issue | Added with no `render()` call | Added later |
| --- | --- | --- | --- |
| 18.0.0 | broken | broken | broken |
| 18.1.0 | works | works | works |
| current `develop` | works | works | works |

On 18.0.0 the grid scrolled to a blank area and kept rendering rows 1 to 11. On 18.1.0 the same scroll moved the rendered rows to 41 to 56, and the column headers moved with the body.

The third column matters most for framework users: the grid settles even if you never call `render()`. Adding the container to the page is a size change that Handsontable already reacts to, and that redraw fixes the scrolling on its own.

### About the workarounds

You no longer need `updateSettings({})` or `wtOverlays.updateMainScrollableElements()`. Both of them worked by forcing that decision to be made again, which now happens by itself. `wtOverlays` is internal API, so I recommend removing that call once you upgrade.

There are also automated tests covering this case now, so it cannot come back unnoticed.

Sorry this took so long. The changelog entry for the fix describes the symptoms but does not link back to this issue, which is why it stayed open after the release.

Closing as resolved.
